config/jobs/kubernetes/sig-node/containerd.yaml

```diff
@@ -879,3 +879,33 @@ periodics:
     testgrid-dashboards: sig-node-containerd
     testgrid-tab-name: cgroup-systemd-containerd-node-e2e
     testgrid-alert-email: kubernetes-sig-node-test-failures@googlegroups.com
+- interval: 12h
+  name: ci-cos-cgroupv2-containerd-e2e-gce
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+    preset-e2e-cos-containerd: "true"
+  spec:
+    containers:
+    - args:
+      - --repo=github.com/containerd/containerd=master
+      - --timeout=70
+      - --scenario=kubernetes_e2e
+      - --
+      - --check-leaked-resources
+      - --env=ENABLE_POD_SECURITY_POLICY=true
+      - --env=KUBE_NODE_EXTRA_METADATA=user-data=/workspace/github.com/containerd/containerd/test/e2e/node.yaml,containerd-configure-sh=/workspace/github.com/containerd/containerd/cluster/gce/configure.sh,containerd-env=/workspace/test-infra/jobs/e2e_node/containerd/containerd-master/cgroupv2/env-cgroupv2
+      - --extract=ci/latest
+      - --gcp-node-image=gci
+      - --gcp-nodes=4
+      - --gcp-zone=us-west1-b
+      - --ginkgo-parallel=30
+      - --image-family=cos-beta
+      - --image-project=cos-cloud
+      - --provider=gce
+      - --test_args=--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
```

Comment:
I think we need to also figure out how to enable systemd cgroup driver in k/k startup scripts.

Because for cgroupv2, we are using systemd cgroup driver -- https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/containerd/containerd-master/cgroupv2/env-cgroupv2#L10

used here
https://github.com/containerd/containerd/blob/main/contrib/gce/configure.sh#L180
https://github.com/containerd/containerd/blob/main/contrib/gce/configure.sh#L209

But we also need to ensure kubelet will use systemd cgroup driver with `--cgroup-driver=systemd`.

Reply:
you might be able to set `NODE_KUBELET_TEST_ARGS`
https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L782

but it looks like it's already set here - https://github.com/kubernetes/test-infra/blob/68c0582bee9da1c995249839ff29cd1088952b03/config/jobs/kubernetes/sig-node/containerd.yaml#L22

so I'm not sure, but maybe there is way to override it for this specific job and set it to something like `--runtime-cgroups=/system.slice/containerd.service --cgroup-driver=systemd`

Reply:
Thanks for the detail information. Yeah, `NODE_KUBELET_TEST_ARGS` should work by specifying the thing like ` - --env=NODE_KUBELET_TEST_ARGS=--cgroup-driver=systemd`. I didn't include the `--runtime-cgroups=/system.slice/containerd.service` as it is already specified in `KUBELET_TEST_ARGS` and be used to construct common kubelet args for both master and nodes: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L742.

Reply:
hmm, I'm not 100% sure but I would assume that if you specify `--env=NODE_KUBELET_TEST_ARGS`, it will override the environment variable used above (i.e. [here](https://github.com/kubernetes/test-infra/blob/2ff402f8b64bb15323bec3b72c0182624c758e47/config/jobs/kubernetes/sig-node/containerd.yaml#L21-L22). 

So I think you'll need to specify both here to ensure `runtime-cgroups` is still passed as a flag, something like

```
--env=NODE_KUBELET_TEST_ARGS=runtime-cgroups=/system.slice/containerd.service --cgroup-driver=systemd
```

Reply:
The `NODE_KUBELET_TEST_ARGS` is not same as `KUBELET_TEST_ARGS` (missing`NODE`) used above actually([here](https://github.com/kubernetes/test-infra/pull/22130/commits/8a193964a1fd038ed257cf5599b9d98ab41ad112#R21))
so it won't get override. `NODE_KUBELET_TEST_ARGS` only appends to kubelet flags for node, `KUBELET_TEST_ARGS` is used for both master and nodes.

Reply:
oh, I see, didn't notice the difference there :) 

Thinking about it some more though, I think we want this to be in `KUBELET_TEST_ARGS` actually.

With the current config, both the master and node will be running cgroupv2 and we want to ensure that the kubelet running on the master (running any static pods) is also using systemd cgroup driver, just like the nodes.

So I think we'll want 

```
--env=KUBELET_TEST_ARGS=runtime-cgroups=/system.slice/containerd.service --cgroup-driver=systemd
```

Reply:
I see. If we want master also running in cgroupv2, `KUBE_MASTER_EXTRA_METADATA` needs to be override as well. Will update it

Reply:
Thanks, nice catch, I think it'll best if we test cgroupv2 on both master and node in these cluster E2E tests.